src/pe/export.rs

```diff
@@ -195,47 +222,64 @@ impl<'a, 'b> scroll::ctx::TryFromCtx<'a, ExportCtx<'b>> for Export<'a> {
+            name_offset.and_then(|offset| bytes.pread::<&str>(offset).ok())
+        });
+
+        if let Some(ordinal) = ordinals.get(idx) {
+            if let Some(rva) = addresses.get(*ordinal as usize) {
+                match *rva {
+                    ExportRVA(rva) => {
+                        let rva = rva as usize;
+                        let offset = utils::find_offset_or(rva, sections, &format!("cannot map RVA ({:#x}) of export ordinal {} into offset", rva, ordinal))?;
+                        Ok((Export { name, offset, rva, reexport: None }, 0))
+                    },
+
+                    ForwarderRVA(rva) => {
+                        let rva = rva as usize;
+                        let offset = utils::find_offset_or(rva, sections, &format!("cannot map RVA ({:#x}) of export ordinal {} into offset", rva, ordinal))?;
+                        let reexport = Reexport::parse(bytes, offset)?;
+                        Ok((Export { name, offset, rva, reexport: Some(reexport) }, 0))
+                    }
+                }
             }
+            else {
+                Err(error::Error::Malformed(format!("cannot get RVA of export ordinal {}", ordinal)))
+            }
+        }
+        else {
+            Err(error::Error::Malformed(format!("cannot get ordinal of export name entry {}", idx)))
         }
     }
 }
 
 impl<'a> Export<'a> {
     pub fn parse(bytes: &'a [u8], export_data: &ExportData, sections: &[section_table::SectionTable]) -> error::Result<Vec<Export<'a>>> {
-        let pointers = &export_data.export_name_pointer_table;
-        let addresses = &export_data.export_address_table;
-        let ordinals = &export_data.export_ordinal_table;
-        //let ordinal_base = export_data.export_directory_table.ordinal_base as usize;
-        let mut exports = Vec::with_capacity(pointers.len());
-        for (idx, ptr) in pointers.iter().enumerate() {
-            let export = bytes.pread_with(0, ExportCtx { ptr: *ptr, idx: idx, sections: sections, addresses: addresses, ordinals: ordinals })?;
-            exports.push(export);
+        let number_of_name_pointers = export_data.export_directory_table.number_of_name_pointers;
+
+        let mut addresses = &Vec::new();
+        if let Some(ref table) = export_data.export_address_table {
+            addresses = table;
+        }
+
+        let mut ordinals = &Vec::new();
+        if let Some(ref table) = export_data.export_ordinal_table {
+            ordinals = table;
         }
+
+        let mut exports = Vec::with_capacity(number_of_name_pointers as usize);
+        for idx in 0..number_of_name_pointers {
+            let ptr = export_data.export_name_pointer_table.as_ref().map_or(None, |pointers| {
+                pointers.get(idx as usize).map(|v| *v)
+            });
+
+            if let Ok(export) = bytes.pread_with(0, ExportCtx { ptr: ptr, idx: idx as usize, sections: sections, addresses: addresses, ordinals: ordinals }) {
```

Comment:
Great usecase for type punning/field initializer syntax, e.g., `ExportCtx { ptr, sections, addresses, ordinals: idx: idx as usize }`